### ISSUE TITLE
Úkol č. 4

### ISSUE THREAD

**Issue by KaterinaVokounova** [KaterinaVokounova/Python2023_Ukoly#4]
Ahoj Martine, 

posílám čtvrtý úkol. Dal mi docela zabrat a asi ne všechno mám správně. Kde jsem si nebyla jistá, tak jsem ti tam přidala poznámku.  Importy jsem nedala na začátek, ale zvolila jsem tu cestu, že jsem je přidávala průběžně. Pomáhá mi to si ujasnit a víc zapamatovat, k čemu jednotlivé importy jsou. 

Tady je odkaz na řešení: https://github.com/KaterinaVokounova/Python2023_Ukoly/blob/4d09303acd326a6be14f457bc76091d289063874/ukol_04.py

A tady na strom, protože jsem nepracovala v Jupyter: https://github.com/KaterinaVokounova/Python2023_Ukoly/blob/4d09303acd326a6be14f457bc76091d289063874/tree_ukol_04.png

Moc díky.

**Comment by M-Kovar** [comment]
Ahoj Katko,

náhodou pěkný:), máš to pečlivě a správně zpracované a píšu splněno. Tímto jsi jako první z našeho týmu posbírala kompletní kolekci všech pěti splněných úkolů, takže gratuluju:-). Pár komentářů:

- V těch importech už je chaos, souhlasím, že takhle je to přehlednější.
- Když se podívám na matici záměn, rozložení dat se zdá být dost nerovnoměrné - dataset obsahuje mnohonásobně víc případů "no" než "yes". Možná by bylo vhodné použít pro `train_test_split()` parametr `stratify=y`, který by zajistil rovnoměrné rozložení případů yes/no v trénovací a testovací sadě. Ale to je jen taková moje neotestovaná hypotéza:).
- Příprava dat pro KNN: Jak říkáš, do stromu normalizace/scaling není potřeba, ale do KNN ano, takže Tvůj postup je správný. Dívám se do Jirkova vzorového řešení a zdá se, že do něj přidával scaling dodatečně. A jestli koukám správně, tak v řešení už výsledná čísla pro KNN i SVC taky vychází zhruba jako Tobě, takže dobrý.
- Hledání optimálního počtu sousedů u KNN: for cyklem vyšlo 23, přes `GridSearchCV()` vyšlo 19. To se může stát, `GridSearchCV()` bych věřil víc, jelikož díky cross-validaci rozděluje data na train/test a provádí KNN vícekrát, zatímco for cyklus dává výsledek "na první dobrou".
- Optimální `max_depth` stromu: Jo, Jirkovi taky vyšlo 8, takže dobrý :D.

Tak díky za spolupráci a přeju hodně datových i nedatových úspěchů do budoucna:).

**Comment by KaterinaVokounova** [comment]
Ahoj Martine,

znovu moc díky za opravu. A hlavně Ti chci poděkovat za to, žes byl úplně
skvělý kouč,  že ses nám tak věnoval, opravoval prakticky hned úkoly ve
svém volném čase, dával super zpětnou vazbu,.... Dělala jsem paralelně
ještě jeden kurz u Czechitas, tak mám teď tendenci trochu srovnávat:-D. .
Statistika i machine learning byl pro mě úplně nový svět, ale tys nás tím
perfektně provedl. Tak ještě jednou moc díky a přeju i Tobě spoustu
úspěchů  a dobré karmy za tuhle práci pro komunitu:-).

Katka

ne 25. 6. 2023 v 16:06 odesílatel Martin Kovar ***@***.***>
napsal:
